### DISCUSSION TITLE
Why does the RLE stop on each line?

### DISCUSSION THREAD

**Discussion by cdt-eth** [nounsDAO/nouns-monorepo#401] [Solidity / smart contracts] [answered]
Is there any particular reason why the run-length encoding (RLE) stops its run on each line? 

e.g. if there is transparency on the end of one line and on the start of the next it stores it as (length-number, color-number) 3 0 and 4 0 instead of say 7 0?

**Comment by solimander** [comment, marked as answer]
Simplicity, mostly. The compression was good enough to lower the deployment cost into a reasonable range, and it was simpler to map a single `[pixel_length, color_index]` tuple to a single rect when drawing the SVG. We didn't feel the need to over-optimize. The renderer, including storage of all parts, is upgradable, so the algorithm can be improved over time. In fact, multi-line RLE has already been implemented in a separate branch.
